docs/cudf/source/developer_guide/ARCHITECTURE.md

```diff
@@ -0,0 +1,199 @@
+# cuDF Architecture
```

Comment:
Is "ARCHITECTURE.md" a standard thing in other projects? My first thought is that this was going to be a document related to supported GPU architectures, so that tells me it's a bit of an overloaded term.

How about `LIBRARY_DESIGN.md`?

Reply:
`ARCHITECTURE.md` is a fairly popular name for this kind of document - it's not as standardized as names like `CONTRIBUTING.md`, but there are a nontrivial number of projects that use `ARCHITECTURE.md` for this purpose.

https://matklad.github.io/2021/02/06/ARCHITECTURE.md.html
https://github.com/rust-lang/rust-analyzer/blob/master/docs/dev/architecture.md

Reply:
Yeah, while I see your argument about the term being overloaded I'd say [this is a popular enough naming convention to stick with](https://github.com/search?p=1&q=filename%3Aarchitecture.md&type=Code) (fair warning: this search finds enough examples that once I started clicking through them to see if they were real (and they mostly are) I got rate limited by Github :joy:)

Reply:
Could we use lowercase though? The examples in the search seem to use both. Someone looking to add a new `.md` file might wonder whether to use `lowercase.md` or `UPPERCASE.md`.

Reply:
I'm +1 to `library-design.md` FWIW but ultimately leave it to Vyas.

Reply:
I do like the name `library-design.md` more, especially since this is not a top-level document in the repo but instead exists as part of the Python cuDF docs.

Reply:
I don't feel that strongly about this. I'll take 3 votes for `library_design.md` as good enough to switch.